File: public/rbxl/macOS/x64/0.726.0.7261140/offsets.cs
/*
 * Dumped With: roblox-dumper 2.7
 * Created by: upio (upio on Discord) & Jonah (jonahw on Discord)
 * Github: https://github.com/notpoiu/roblox-dumper (forked)
 * Roblox Version: 0.726.0.7261140
 * Architecture: x64
 * Time Taken: 6430 ms (6.430000 seconds)
 * Total Offsets: 276
 */

using System;

namespace RobloxOffsets
{
    public static class Metadata
    {
        public const string RobloxVersion = "0.726.0.7261140";
        public const string Architecture = "x64";
    }

    public static class Atmosphere
    {
        public const ulong Color = 0xD0;
        public const ulong Decay = 0xDC;
        public const ulong Density = 0xE8;
        public const ulong Glare = 0xEC;
        public const ulong Haze = 0xF0;
        public const ulong Offset = 0xF4;
    }

    public static class Attribute
    {
        public const ulong Key = 0x0;
        public const ulong Size = 0x58;
        public const ulong Value = 0x18;
    }

    public static class BasePart
    {
        public const ulong CastShadow = 0xED;
        public const ulong Color3 = 0x184;
        public const ulong Locked = 0xEE;
        public const ulong Massless = 0xEF;
        public const ulong Primitive = 0x138;
        public const ulong Reflectance = 0xE4;
        public const ulong Shape = 0x1A1;
        public const ulong Transparency = 0xE8;
    }

    public static class BloomEffect
    {
        public const ulong Intensity = 0xCC;
        public const ulong Size = 0xD0;
        public const ulong Threshold = 0xD4;
    }

    public static class ByteCode
    {
        public const ulong Pointer = 0x10;
        public const ulong Size = 0x18;
    }

    public static class Camera
    {
        public const ulong CFrame = 0xF0;
        public const ulong FieldOfView = 0x158;
        public const ulong Position = 0x114;
        public const ulong Rotation = 0xF0;
        public const ulong ViewportInt16 = 0x2A4;
        public const ulong ViewportSize = 0x2E0;
    }

    public static class CharacterMesh
    {
        public const ulong BaseTextureId = 0xE0;
        public const ulong BodyPart = 0x148;
        public const ulong MeshId = 0x108;
        public const ulong OverlayTextureId = 0x130;
    }

    public static class DataModel
    {
        public const ulong CreatorId = 0x188;
        public const ulong GameId = 0x190;
        public const ulong GameLoaded = 0xA38;
        public const ulong JobId = 0x148;
        public const ulong PlaceId = 0x198;
        public const ulong Workspace = 0x168;
    }

    public static class FakeDataModel
    {
        public const ulong RealDataModel = 0x1D8;
    }

    public static class GuiBase2D
    {
        public const ulong AbsolutePosition = 0x10C;
        public const ulong AbsoluteRotation = 0x180;
        public const ulong AbsoluteSize = 0x118;
    }

    public static class GuiObject
    {
        public const ulong Active = 0x5A8;
        public const ulong AnchorPoint = 0x558;
        public const ulong AutomaticSize = 0x560;
        public const ulong BackgroundColor3 = 0x540;
        public const ulong BackgroundTransparency = 0x564;
        public const ulong BorderColor3 = 0x54C;
        public const ulong BorderMode = 0x568;
        public const ulong BorderSizePixel = 0x56C;
        public const ulong ClipsDescendants = 0x5A9;
        public const ulong GuiState = 0x578;
        public const ulong Interactable = 0x5AB;
        public const ulong LayoutOrder = 0x580;
        public const ulong Position = 0x510;
        public const ulong Rotation = 0x180;
        public const ulong Selectable = 0x5AC;
        public const ulong SelectionOrder = 0x59C;
        public const ulong Size = 0x530;
        public const ulong SizeConstraint = 0x5A0;
        public const ulong Visible = 0x5AD;
        public const ulong ZIndex = 0x5A4;
    }

    public static class Humanoid
    {
        public const ulong AutoJumpEnabled = 0x1D0;
        public const ulong AutoRotate = 0x1D1;
        public const ulong AutomaticScalingEnabled = 0x1D2;
        public const ulong BreakJointsOnDeath = 0x1D3;
        public const ulong CameraOffset = 0x130;
        public const ulong DisplayDistanceType = 0x17C;
        public const ulong EvaluateStateMachine = 0x1D4;
        public const ulong Health = 0x184;
        public const ulong HealthDisplayDistance = 0x188;
        public const ulong HealthDisplayType = 0x18C;
        public const ulong HipHeight = 0x190;
        public const ulong JumpHeight = 0x19C;
        public const ulong JumpPower = 0x1A0;
        public const ulong MaxHealth = 0x1A4;
        public const ulong MaxSlopeAngle = 0x1A8;
        public const ulong NameDisplayDistance = 0x1AC;
        public const ulong NameOcclusion = 0x1B0;
        public const ulong RequiresNeck = 0x1D9;
        public const ulong RigType = 0x1BC;
        public const ulong SeatPart = 0x110;
        public const ulong Sit = 0x1DA;
        public const ulong TargetPoint = 0x154;
        public const ulong UseJumpPower = 0x1DC;
        public const ulong WalkSpeed = 0x1CC;
        public const ulong WalkSpeedCheck = 0x3B4;
        public const ulong WalkToPoint = 0x16C;
    }

    public static class InputObject
    {
        public const ulong MousePosition = 0xEC;
    }

    public static class Instance
    {
        public const ulong ChildrenEnd = 0x8;
        public const ulong ChildrenStart = 0x78;
        public const ulong ClassDescriptor = 0x18;
        public const ulong ClassName = 0x8;
        public const ulong ComponentMap = 0x38;
        public const ulong Name = 0xB0;
        public const ulong Parent = 0x70;
    }

    public static class Lighting
    {
        public const ulong Ambient = 0xE0;
        public const ulong Atmosphere = 0x1E8;
        public const ulong Brightness = 0x128;
        public const ulong ClockTime = 0x1B8;
        public const ulong ColorShift_Bottom = 0xEC;
        public const ulong ColorShift_Top = 0xF8;
        public const ulong EnvironmentDiffuseScale = 0x12C;
        public const ulong EnvironmentSpecularScale = 0x130;
        public const ulong ExposureCompensation = 0x134;
        public const ulong FogColor = 0x104;
        public const ulong FogEnd = 0x13C;
        public const ulong FogStart = 0x140;
        public const ulong OutdoorAmbient = 0x110;
        public const ulong ShadowSoftness = 0x148;
        public const ulong Sky = 0x1D8;
    }

    public static class LightingParameters
    {
        public const ulong GeographicLatitude = 0x194;
        public const ulong LightColor = 0x160;
        public const ulong LightDirection = 0x16C;
        public const ulong SkyAmbient = 0x154;
        public const ulong SkyAmbient2 = 0x198;
        public const ulong Source = 0x178;
        public const ulong TrueMoonPosition = 0x188;
        public const ulong TrueSunPosition = 0x17C;
    }

    public static class LocalScript
    {
        public const ulong Bytecode = 0x198;
        public const ulong Hash = 0x10;
    }

    public static class MaterialColors
    {
        public const ulong Asphalt = 0x30;
        public const ulong Basalt = 0x27;
        public const ulong Brick = 0xF;
        public const ulong Cobblestone = 0x33;
        public const ulong Concrete = 0xC;
        public const ulong CrackedLava = 0x2D;
        public const ulong Glacier = 0x1B;
        public const ulong Grass = 0x6;
        public const ulong Ground = 0x2A;
        public const ulong Ice = 0x36;
        public const ulong LeafyGrass = 0x39;
        public const ulong Limestone = 0x3F;
        public const ulong Mud = 0x24;
        public const ulong Pavement = 0x42;
        public const ulong Rock = 0x18;
        public const ulong Salt = 0x3C;
        public const ulong Sand = 0x12;
        public const ulong Sandstone = 0x21;
        public const ulong Slate = 0x9;
        public const ulong Snow = 0x1E;
        public const ulong WoodPlanks = 0x15;
    }

    public static class MeshPart
    {
        public const ulong MeshId = 0x2E0;
        public const ulong TextureId = 0x308;
    }

    public static class Model
    {
        public const ulong PrimaryPart = 0x238;
    }

    public static class ModuleScript
    {
        public const ulong Bytecode = 0x140;
        public const ulong Hash = 0x10;
    }

    public static class MouseService
    {
        public const ulong InputObject = 0x118;
    }

    public static class Player
    {
        public const ulong AccountAge = 0x304;
        public const ulong DisplayName = 0x138;
        public const ulong HealthDisplayDistance = 0x334;
        public const ulong LocaleId = 0x650;
        public const ulong ModelInstance = 0x388;
        public const ulong NameDisplayDistance = 0x344;
        public const ulong Team = 0x288;
        public const ulong TeamColor = 0x350;
        public const ulong UserId = 0x2B0;
    }

    public static class Players
    {
        public const ulong LocalPlayer = 0x138;
    }

    public static class Primitive
    {
        public const ulong AssemblyAngularVelocity = 0x104;
        public const ulong AssemblyLinearVelocity = 0xF8;
        public const ulong CFrame = 0xC8;
        public const ulong Material = 0x23E;
        public const ulong Orientation = 0xC8;
        public const ulong Position = 0xEC;
        public const ulong PrimitiveFlags = 0x1B6;
        public const ulong Rotation = 0xC8;
        public const ulong Size = 0x1B8;
    }

    public static class PrimitiveFlags
    {
        public const ulong Anchored = 0x2;
        public const ulong CanCollide = 0x8;
        public const ulong CanQuery = 0x20;
        public const ulong CanTouch = 0x10;
    }

    public static class ProximityPrompt
    {
        public const ulong ActionText = 0xC8;
        public const ulong Enabled = 0x13E;
        public const ulong HoldDuration = 0x128;
        public const ulong KeyboardKeyCode = 0x12C;
        public const ulong MaxActivationDistance = 0x130;
        public const ulong ObjectText = 0xE0;
        public const ulong RequiresLineOfSight = 0x13F;
    }

    public static class RenderView
    {
        public const ulong LightingValid = 0x170;
        public const ulong SkyboxValid = 0x28D;
    }

    public static class Seat
    {
        public const ulong Occupant = 0x208;
    }

    public static class Sky
    {
        public const ulong MoonAngularSize = 0x21C;
        public const ulong MoonTextureId = 0xE0;
        public const ulong SkyboxBk = 0x108;
        public const ulong SkyboxDn = 0x130;
        public const ulong SkyboxFt = 0x158;
        public const ulong SkyboxLf = 0x180;
        public const ulong SkyboxOrientation = 0x210;
        public const ulong SkyboxRt = 0x1A8;
        public const ulong SkyboxUp = 0x1D0;
        public const ulong StarCount = 0x220;
        public const ulong SunAngularSize = 0x224;
        public const ulong SunTextureId = 0x1F8;
    }

    public static class SpecialMesh
    {
        public const ulong MeshId = 0x110;
        public const ulong Offset = 0xD0;
        public const ulong Scale = 0xDC;
        public const ulong TextureId = 0x138;
    }

    public static class TaskScheduler
    {
        public const ulong JobEnd = 0xD8;
        public const ulong JobName = 0x18;
        public const ulong JobStart = 0xD0;
        public const ulong MaxFps = 0xB8;
        public const ulong Pointer = 0x67F3A80;
    }

    public static class Team
    {
        public const ulong TeamColor = 0xD0;
    }

    public static class Terrain
    {
        public const ulong GrassLength = 0x1E0;
        public const ulong MaterialColors = 0x490;
        public const ulong WaterColor = 0x1D0;
        public const ulong WaterReflectance = 0x1E8;
        public const ulong WaterTransparency = 0x1EC;
        public const ulong WaterWaveSize = 0x1F0;
        public const ulong WaterWaveSpeed = 0x1F4;
    }

    public static class TextButton
    {
        public const ulong AutoButtonColor = 0x9C4;
        public const ulong ContentText = 0xDC8;
        public const ulong Font = 0x1098;
        public const ulong LineHeight = 0xEA8;
        public const ulong LocalizedText = 0xDC8;
        public const ulong MaxVisibleGraphemes = 0x10A4;
        public const ulong Modal = 0x9C5;
        public const ulong RichText = 0xFA6;
        public const ulong Selected = 0x9C6;
        public const ulong Text = 0xDC8;
        public const ulong TextColor3 = 0x1080;
        public const ulong TextDirection = 0xF48;
        public const ulong TextScaled = 0xF9E;
        public const ulong TextSize = 0x10AC;
        public const ulong TextStrokeColor3 = 0x108C;
        public const ulong TextStrokeTransparency = 0x10B0;
        public const ulong TextTransparency = 0x10B4;
        public const ulong TextTruncate = 0x10B8;
        public const ulong TextWrapped = 0xFA0;
        public const ulong TextXAlignment = 0x10BC;
        public const ulong TextYAlignment = 0xEF0;
    }

    public static class TextLabel
    {
        public const ulong ContentText = 0xB40;
        public const ulong Font = 0xE10;
        public const ulong LineHeight = 0xC20;
        public const ulong LocalizedText = 0xB40;
        public const ulong MaxVisibleGraphemes = 0xE1C;
        public const ulong RichText = 0xD1E;
        public const ulong Text = 0xB40;
        public const ulong TextColor3 = 0xDF8;
        public const ulong TextDirection = 0xCC0;
        public const ulong TextScaled = 0xD16;
        public const ulong TextSize = 0xE24;
        public const ulong TextStrokeColor3 = 0xE04;
        public const ulong TextStrokeTransparency = 0xE28;
        public const ulong TextTransparency = 0xE2C;
        public const ulong TextTruncate = 0xE30;
        public const ulong TextWrapped = 0xD18;
        public const ulong TextXAlignment = 0xE34;
        public const ulong TextYAlignment = 0xC68;
    }

    public static class Tool
    {
        public const ulong CanBeDropped = 0x450;
        public const ulong Enabled = 0x451;
        public const ulong Grip = 0x420;
        public const ulong GripForward = 0x438;
        public const ulong GripPos = 0x444;
        public const ulong GripRight = 0x420;
        public const ulong GripUp = 0x42C;
        public const ulong ManualActivationOnly = 0x452;
        public const ulong RequiresHandle = 0x453;
        public const ulong Tooltip = 0x408;
    }

    public static class Value
    {
        public const ulong Value = 0xD0;
    }

    public static class VehicleSeat
    {
        public const ulong MaxSpeed = 0x220;
        public const ulong Occupant = 0x200;
        public const ulong SteerFloat = 0x228;
        public const ulong ThrottleFloat = 0x230;
        public const ulong Torque = 0x234;
        public const ulong TurnSpeed = 0x238;
    }

    public static class VisualEngine
    {
        public const ulong FakeDataModel = 0xA80;
        public const ulong Pointer = 0x65012C0;
        public const ulong RenderView = 0xBA8;
        public const ulong ViewMatrix = 0x140;
    }

    public static class Workspace
    {
        public const ulong CurrentCamera = 0x458;
        public const ulong ReadOnlyGravity = 0x958;
        public const ulong World = 0x3B8;
    }

    public static class World
    {
        public const ulong Gravity = 0x208;
        public const ulong Primitives = 0x260;
        public const ulong WorldSteps = 0x618;
    }

} // namespace RobloxOffsets

File: public/rbxl/macOS/x64/0.727.0.7271199/offsets.py
# Dumped With: roblox-dumper 2.7
# Created by: upio (upio on Discord) & Jonah (jonahw on Discord)
# Github: https://github.com/notpoiu/roblox-dumper (forked)
# Roblox Version: 0.727.0.7271199
# Architecture: x64
# Time Taken: 7681 ms (7.681000 seconds)
# Total Offsets: 276

class Offsets:
    ROBLOX_VERSION = "0.727.0.7271199"
    ARCHITECTURE = "x64"

    class Atmosphere:
        Color = 208
        Decay = 220
        Density = 232
        Glare = 236
        Haze = 240
        Offset = 244

    class Attribute:
        Key = 0
        Size = 88
        Value = 24

    class BasePart:
        CastShadow = 237
        Color3 = 388
        Locked = 238
        Massless = 239
        Primitive = 312
        Reflectance = 228
        Shape = 417
        Transparency = 232

    class BloomEffect:
        Intensity = 204
        Size = 208
        Threshold = 212

    class ByteCode:
        Pointer = 16
        Size = 24

    class Camera:
        CFrame = 240
        FieldOfView = 344
        Position = 276
        Rotation = 240
        ViewportInt16 = 676
        ViewportSize = 736

    class CharacterMesh:
        BaseTextureId = 224
        BodyPart = 328
        MeshId = 264
        OverlayTextureId = 304

    class DataModel:
        CreatorId = 392
        GameId = 400
        GameLoaded = 2624
        JobId = 328
        PlaceId = 408
        Workspace = 360

    class FakeDataModel:
        RealDataModel = 464

    class GuiBase2D:
        AbsolutePosition = 268
        AbsoluteRotation = 384
        AbsoluteSize = 280

    class GuiObject:
        Active = 1456
        AnchorPoint = 1376
        AutomaticSize = 1384
        BackgroundColor3 = 1352
        BackgroundTransparency = 1388
        BorderColor3 = 1364
        BorderMode = 1392
        BorderSizePixel = 1396
        ClipsDescendants = 1457
        GuiState = 1408
        Interactable = 1459
        LayoutOrder = 1416
        Position = 1304
        Rotation = 384
        Selectable = 1460
        SelectionOrder = 1444
        Size = 1336
        SizeConstraint = 1448
        Visible = 1461
        ZIndex = 1452

    class Humanoid:
        AutoJumpEnabled = 464
        AutoRotate = 465
        AutomaticScalingEnabled = 466
        BreakJointsOnDeath = 467
        CameraOffset = 304
        DisplayDistanceType = 380
        EvaluateStateMachine = 468
        Health = 388
        HealthDisplayDistance = 392
        HealthDisplayType = 396
        HipHeight = 400
        JumpHeight = 412
        JumpPower = 416
        MaxHealth = 420
        MaxSlopeAngle = 424
        NameDisplayDistance = 428
        NameOcclusion = 432
        RequiresNeck = 473
        RigType = 444
        SeatPart = 272
        Sit = 474
        TargetPoint = 340
        UseJumpPower = 476
        WalkSpeed = 460
        WalkSpeedCheck = 948
        WalkToPoint = 364

    class InputObject:
        MousePosition = 236

    class Instance:
        ChildrenEnd = 8
        ChildrenStart = 120
        ClassDescriptor = 24
        ClassName = 8
        ComponentMap = 56
        Name = 176
        Parent = 112

    class Lighting:
        Ambient = 224
        Atmosphere = 488
        Brightness = 296
        ClockTime = 440
        ColorShift_Bottom = 236
        ColorShift_Top = 248
        EnvironmentDiffuseScale = 300
        EnvironmentSpecularScale = 304
        ExposureCompensation = 308
        FogColor = 260
        FogEnd = 316
        FogStart = 320
        OutdoorAmbient = 272
        ShadowSoftness = 328
        Sky = 472

    class LightingParameters:
        GeographicLatitude = 404
        LightColor = 352
        LightDirection = 364
        SkyAmbient = 340
        SkyAmbient2 = 408
        Source = 376
        TrueMoonPosition = 392
        TrueSunPosition = 380

    class LocalScript:
        Bytecode = 408
        Hash = 16

    class MaterialColors:
        Asphalt = 48
        Basalt = 39
        Brick = 15
        Cobblestone = 51
        Concrete = 12
        CrackedLava = 45
        Glacier = 27
        Grass = 6
        Ground = 42
        Ice = 54
        LeafyGrass = 57
        Limestone = 63
        Mud = 36
        Pavement = 66
        Rock = 24
        Salt = 60
        Sand = 18
        Sandstone = 33
        Slate = 9
        Snow = 30
        WoodPlanks = 21

    class MeshPart:
        MeshId = 736
        TextureId = 776

    class Model:
        PrimaryPart = 568

    class ModuleScript:
        Bytecode = 320
        Hash = 16

    class MouseService:
        InputObject = 280

    class Player:
        AccountAge = 780
        DisplayName = 312
        HealthDisplayDistance = 832
        LocaleId = 1632
        ModelInstance = 912
        NameDisplayDistance = 848
        Team = 648
        TeamColor = 860
        UserId = 688

    class Players:
        LocalPlayer = 312

    class Primitive:
        AssemblyAngularVelocity = 260
        AssemblyLinearVelocity = 248
        CFrame = 200
        Material = 574
        Orientation = 200
        Position = 236
        PrimitiveFlags = 438
        Rotation = 200
        Size = 440

    class PrimitiveFlags:
        Anchored = 2
        CanCollide = 8
        CanQuery = 32
        CanTouch = 16

    class ProximityPrompt:
        ActionText = 200
        Enabled = 318
        HoldDuration = 296
        KeyboardKeyCode = 300
        MaxActivationDistance = 304
        ObjectText = 224
        RequiresLineOfSight = 319

    class RenderView:
        LightingValid = 544
        SkyboxValid = 653

    class Seat:
        Occupant = 520

    class Sky:
        MoonAngularSize = 540
        MoonTextureId = 224
        SkyboxBk = 264
        SkyboxDn = 304
        SkyboxFt = 344
        SkyboxLf = 384
        SkyboxOrientation = 528
        SkyboxRt = 424
        SkyboxUp = 464
        StarCount = 544
        SunAngularSize = 548
        SunTextureId = 504

    class SpecialMesh:
        MeshId = 272
        Offset = 208
        Scale = 220
        TextureId = 312

    class TaskScheduler:
        JobEnd = 216
        JobName = 1304
        JobStart = 208
        MaxFps = 184
        Pointer = 109380848

    class Team:
        TeamColor = 208

    class Terrain:
        GrassLength = 480
        MaterialColors = 1168
        WaterColor = 464
        WaterReflectance = 488
        WaterTransparency = 492
        WaterWaveSize = 496
        WaterWaveSpeed = 500

    class TextButton:
        AutoButtonColor = 2508
        ContentText = 3536
        Font = 4256
        LineHeight = 3760
        LocalizedText = 3536
        MaxVisibleGraphemes = 4268
        Modal = 2509
        RichText = 4014
        Selected = 2510
        Text = 3536
        TextColor3 = 4232
        TextDirection = 3920
        TextScaled = 4006
        TextSize = 4276
        TextStrokeColor3 = 4244
        TextStrokeTransparency = 4280
        TextTransparency = 4284
        TextTruncate = 4288
        TextWrapped = 4008
        TextXAlignment = 4292
        TextYAlignment = 3832

    class TextLabel:
        ContentText = 2888
        Font = 3608
        LineHeight = 3112
        LocalizedText = 2888
        MaxVisibleGraphemes = 3620
        RichText = 3366
        Text = 2888
        TextColor3 = 3584
        TextDirection = 3272
        TextScaled = 3358
        TextSize = 3628
        TextStrokeColor3 = 3596
        TextStrokeTransparency = 3632
        TextTransparency = 3636
        TextTruncate = 3640
        TextWrapped = 3360
        TextXAlignment = 3644
        TextYAlignment = 3184

    class Tool:
        CanBeDropped = 1128
        Enabled = 1129
        Grip = 1080
        GripForward = 1104
        GripPos = 1116
        GripRight = 1080
        GripUp = 1092
        ManualActivationOnly = 1130
        RequiresHandle = 1131
        Tooltip = 1056

    class Value:
        Value = 208

    class VehicleSeat:
        MaxSpeed = 544
        Occupant = 512
        SteerFloat = 552
        ThrottleFloat = 560
        Torque = 564
        TurnSpeed = 568

    class VisualEngine:
        FakeDataModel = 2688
        Pointer = 106286320
        RenderView = 2984
        ViewMatrix = 320

    class Workspace:
        CurrentCamera = 1136
        ReadOnlyGravity = 2368
        World = 976

    class World:
        Gravity = 520
        Primitives = 608
        WorldSteps = 1560

